src/main/java/sparksoniq/jsoniq/runtime/iterator/RuntimeIterator.java

```diff
@@ -45,12 +45,15 @@
 
 	private static final long serialVersionUID = 1L;
 	protected static final String FLOW_EXCEPTION_MESSAGE = "Invalid next() call; ";
-    private final IteratorMetadata metadata;
+    private IteratorMetadata metadata;
     protected boolean _hasNext;
```

Comment:
this can/should transient, set to the same as what the regular constructor does upon reading.